packages/wallets/src/components/WalletHistory.tsx

```diff
@@ -249,6 +263,17 @@ export default function WalletHistory(props: Props) {
         metadata={metadata}
         expandedCellShift={1}
         uniqueField="name"
+        onToggleExpand={async (rowId: string, expanded: boolean) => {
+          const memos = (await getTransactionMemo({ transactionId: rowId.split('-')[0] })).data;
+          if (expanded) {
+            const newTransactions = transactions.map((transaction) =>
+              transaction.name === rowId.split('-')[0] && memos && Object.keys(memos).length > 0
+                ? { ...transaction, memos }
+                : transaction
+            );
+            setTransactions(newTransactions);
```

Comment:
This seems really inefficient. Instead of replacing the entire collection of transactions with a modified version, what if we just maintained a map of transaction IDs and their memos (previously fetched). Then when we look for the memo on a row, try to pull it from the row first, and if not there, try to pull it from the map.

Reply:
This is me being lazy :)
Fixed.